File: src/std/writeAll.zig
const std = @import("std");
const Slice = @import("Slice.zig").Slice;

pub export fn writeAll(f: *std.c.FILE, str: Slice(u8)) void {
    var total_written: usize = 0;
    var written = std.c.fwrite(str.ptr + total_written, 1, str.len - total_written, f);
    while (written != 0) {
        total_written += written;
        written = std.c.fwrite(str.ptr + total_written, 1, str.len - total_written, f);
    }
}

File: src/c/src/std/writeAll.c
#ifndef WRITE_ALL_H
#define WRITE_ALL_H

#include "Result.h"
#include "defs.h"
#include <stdio.h>

static void writeAll(FILE *f, Slice(char) str) {
  size_t written = 0;
  size_t total_written = 0;
  while ((written =
              fwrite(str.ptr + total_written, 1, str.len - total_written, f))) {
    total_written += written;
  }
}

#endif /* WRITE_ALL_H */
